lib/ddtrace/tracer.rb

```diff
@@ -288,7 +288,14 @@ def trace(name, options = {})
         # It's not a problem since we re-raise it afterwards so for example a
         # SignalException::Interrupt would still bubble up.
         rescue Exception => e
-          (options[:on_error] || DEFAULT_ON_ERROR).call(span, e)
+          valid_error = options[:on_error] || DEFAULT_ON_ERROR
+          begin
+            valid_error.call(span, e) if valid_error.respond_to?(:call)
+          rescue
+            Datadog.logger.debug("Failed to capture :on_error: #{e}")
+            DEFAULT_ON_ERROR.call(span, e)
+          end
```

Comment:
```suggestion
          if (on_error_handler = options[:on_error] && on_error_handler.respond_to?(:call))
            begin
              on_error_handler.call(span, e)
            rescue
              Datadog.logger.debug("custom on_error handler failed, falling back to default")
              DEFAULT_ON_ERROR.call(span, e)
            end
          else
            Datadog.logger.debug("Custom on_error handler must be a callable, falling back to default") if on_error_handler
            DEFAULT_ON_ERROR.call(span, e)
          end
```

Reply:
just cleaned this up a bit so that we make sure we still fall back to `DEFAULT_ON_ERROR.call(span, e)` if `:on_error` is not a callable. Current behavior was simply not calling any handler in that case.

Additionally just cleaned up the variable names and debug logs a bit to be more appropriate